neurodamus/connection_manager.py

```diff
@@ -715,97 +714,99 @@ def target_gids(gids):
             #  - Blocks increase as a geometric progression for handling very large sets
 
-            metype_estimate = Counter()
+            metype_estimate = 0
             sampled_gids_count = 0
 
             for start, stop, in gen_ranges(me_gids_count, BLOCK_BASE_SIZE, block_increase_rate=1.1):
-                logging.debug("Processing range %d:%d", start, stop)
+                logging.debug(" - Processing range %d:%d", start, stop)
                 block_len = stop - start
-                sample = me_gids[start:(start + SAMPLES_PER_BLOCK)]
+                sample = me_gids[start:(start + SAMPLED_CELLS_PER_BLOCK)]
                 sample_len = len(sample)
                 if not sample_len:
                     continue
-                try:
-                    if self.CONNECTIONS_TYPE == ConnectionTypes.Synaptic:
-                        sample_counts = self._synapse_reader.get_counts(
-                            sample, group_by="syn_type_id")
+
+                sample_counts = self._synapse_reader.get_conn_counts(sample)
+                total_connections = 0
+                selected_conn_count = 0
+                new_conn_count = 0  # Let's count those which were not "created" before
+                new_syns_count = 0
+
+                for tgid, tgid_conn_counts in sample_counts.items():
+                    total_connections += len(tgid_conn_counts)
+                    if src_target:
+                        conn_sgids = numpy.fromiter(tgid_conn_counts.keys(), dtype="uint32")
+                        sgids_in_target = conn_sgids[src_target.contains(conn_sgids, raw_gids=True)]
                     else:
-                        sample_counts = self._synapse_reader.get_counts(sample)
-                        sample_counts = {self.CONNECTIONS_TYPE: sample_counts}
-                except SonataError as e:
-                    logging.warning("Error while getting synapse counts: %s", e)
-                    logging.warning("Skipping range %d:%d", start, stop)
-                    continue
-                logging.debug("Gids: %s... Types: %s", sample[:10], sample_counts)
-                logging.debug("Average syn/cell: %.2f", sum(sample_counts.values()) / sample_len)
+                        sgids_in_target = tgid_conn_counts.keys()
+
+                    selected_conn_count += len(sgids_in_target)
+                    tgid_connected_sgids = self._dry_run_conns[tgid]
+
+                    for sgid in sgids_in_target:
+                        if sgid not in tgid_connected_sgids:
+                            new_conn_count += 1
+                            new_syns_count += tgid_conn_counts[sgid]
+                            tgid_connected_sgids.add(int(sgid))
+
+                logging.debug(" - Connections (new/selected/total): %d / %d / %d ",
+                              new_conn_count, selected_conn_count, total_connections)
+                block_syns_per_cell = new_syns_count / sample_len
```

Comment:
Is there a check for `sample_len` to not be zero?

Reply:
A few lines up we have a check `if not sample_len: continue`